### PR TITLE
Add functions to read WavelengthFile, LineListFile, ReferenceSpectrum

### PR DESCRIPTION
**Pull request by ziggyman**
and writePfsArm,
include these functions in reduceArc(RefSpec)Task.py

### CODE REVIEW

File: python/pfs/drp/stella/utils.py [hunk 1]
```diff
@@ -128,3 +131,60 @@ def createPfsFiberTrace(dataId, fiberTraceSet, nRows):
...
+    traceIdsTemp = np.ndarray(shape=(len(tbdata)), dtype='int')
+    xCenters = np.ndarray(shape=(len(tbdata)), dtype='float32')
+    yCenters = np.ndarray(shape=(len(tbdata)), dtype='float32')
+    wavelengths = np.ndarray(shape=(len(tbdata)), dtype='float32')
+    traceIdsTemp[:] = tbdata[:]['fiberNum']
+    traceIds = traceIdsTemp.astype('int32')
+    wavelengths[:] = tbdata[:]['pixelWave']
+    xCenters[:] = tbdata[:]['xc']
+    yCenters[:] = tbdata[:]['yc']
+    return {'xCenters': xCenters, 'yCenters': yCenters, 'wavelengths': wavelengths, 'traceIds': traceIds}
```
Comment (jdswinbank) [inline python/pfs/drp/stella/utils.py]: Why are you returning a `dict` here? I think any of `pipeBase.struct`, `collections.NamedTuple` or just a plain old `tuple` would be more idiomatic.

File: python/pfs/drp/stella/reduceArcTask.py [hunk 2]
```diff
@@ -76,6 +75,18 @@ def run(self, expRefList, butler, wLenFile=None, lineList=None, immediate=True):
         self.log.info('wLenFile = %s' % wLenFile)
         self.log.info('lineList = %s' % lineList)
 
+        """ read wavelength file """
+        wavelengthFile = readWavelengthFile(wLenFile)
+        traceIds = wavelengthFile['traceIds']
+        wavelengths = wavelengthFile['wavelengths']
+        xCenters = wavelengthFile['xCenters']
+        yCenters = wavelengthFile['yCenters']
+        traceIdsUnique = np.unique(traceIds)
```
Comment (jdswinbank) [inline python/pfs/drp/stella/reduceArcTask.py]: This unpacking is tiresome. I comment on this below, but you could write this much more concisely if `readWavelengthFile` returned a `NamedTuple`.

File: python/pfs/drp/stella/reduceArcTask.py [hunk 3]
```diff
@@ -76,6 +75,18 @@ def run(self, expRefList, butler, wLenFile=None, lineList=None, immediate=True):
...
+        self.log.debug('traceIdsUnique = %s' % (np.array_str(traceIdsUnique)))
+
+        """ read line list """
+        lineListArr = readLineListFile(lineList)
```
Comment (jdswinbank) [inline python/pfs/drp/stella/reduceArcTask.py]: Again moving outside the loop — same consideration as earlier. Great if it's safe!

File: python/pfs/drp/stella/reduceArcTask.py [hunk 4]
```diff
@@ -149,7 +138,6 @@ def run(self, expRefList, butler, wLenFile=None, lineList=None, immediate=True):
                 spec.setITrace(iTraces[i])
                 self.log.info('flatFiberTraceSet.getFiberTrace(%d).getITrace() = %d, spec.getITrace() = %d' %(i,flatFiberTraceSet.getFiberTrace(i).getITrace(), spec.getITrace()))
                 specSpec = spec.getSpectrum()
-                self.log.info('yCenters.shape = %d' % yCenters.shape)
```
Comment (jdswinbank) [inline python/pfs/drp/stella/reduceArcTask.py]: Why? I think the log message was probably redundant (I guess it logged the same thing for every loop iteration), but it now seems to be gone entirely (ie not even logged once) without explanation. If you just don't want it any more, I'd have this on a separate commit with a message like "Removed redundant log message".

File: python/pfs/drp/stella/reduceArcRefSpecTask.py [hunk 5]
```diff
@@ -78,6 +77,13 @@ def run(self, expRefList, butler, refSpec=None, lineList=None, immediate=True):
         self.log.debug('refSpec = %s' % refSpec)
         self.log.debug('lineList = %s' % lineList)
 
+        """ read line list """
+        lineListArr = readLineListFile(lineList)
+
+        """ read reference Spectrum """
+        refSpecArr = readReferenceSpectrum(refSpec)
+        self.log.debug('len(refSpecArr) = %d' % len(refSpecArr))
+
```
Comment (ziggyman) [inline python/pfs/drp/stella/reduceArcRefSpecTask.py]: I agree. refSpecArr does not get changed

File: python/pfs/drp/stella/utils.py (hunk 1 above)
Comment (ziggyman) [inline python/pfs/drp/stella/utils.py]: changed readWavelengthFile to return a tuple and then
{{xCenters, wavelengths, traceIds = readWavelengthFile(self.wLenFile)}}

File: python/pfs/drp/stella/reduceArcTask.py (hunk 3 above)
Comment (ziggyman) [inline python/pfs/drp/stella/reduceArcTask.py]: It is safe

File: python/pfs/drp/stella/reduceArcTask.py (hunk 2 above)
Comment (ziggyman) [inline python/pfs/drp/stella/reduceArcTask.py]: see above comment

File: python/pfs/drp/stella/reduceArcTask.py (hunk 4 above)
Comment (ziggyman) [inline python/pfs/drp/stella/reduceArcTask.py]: This line doesn't exist anymore due to previous code changes

**Pull request merged**